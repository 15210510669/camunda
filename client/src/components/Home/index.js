/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

export {default as Home} from './Home';
export {default as Collection} from './Collection';
