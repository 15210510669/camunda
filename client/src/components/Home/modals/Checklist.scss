@import 'shared-styles/shared-styles.scss';

.Checklist {
  font-size: 0.875em;

  .itemsList {
    max-height: calc(100vh - 415px);

    &:not(:only-child) {
      margin-top: 8px;
      padding-top: 8px;
      border-top: 1px solid #d3d3d3;
    }

    @include scrollable;
  }
}
