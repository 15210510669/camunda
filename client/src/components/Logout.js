/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import {useEffect} from 'react';
import {withRouter} from 'react-router-dom';
import {get} from 'request';
import {withErrorHandling} from 'HOC';
import {addNotification} from 'notifications';
import {t} from 'translation';

export function Logout({mightFail, history}) {
  useEffect(() => {
    mightFail(
      get('api/authentication/logout'),
      () => {
        addNotification({text: t('navigation.logoutSuccess')});
        history.replace('/');
      },
      () => {
        addNotification({text: t('navigation.logoutFailed'), type: 'error'});
        history.replace('/');
      }
    );
  }, [mightFail, history]);

  return null;
}

export default withRouter(withErrorHandling(Logout));
