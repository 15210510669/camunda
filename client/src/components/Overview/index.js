export {default as Overview} from './Overview';
export {default as Alerts} from './Alerts';
