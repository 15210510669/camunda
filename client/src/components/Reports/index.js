export {default as Reports} from './Reports';
export {default as Report} from './Report';
export {Filter} from './filter'
