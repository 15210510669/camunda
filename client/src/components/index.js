export {Footer} from './Footer';
export {Header, LogoutButton} from './Header';
export {Home} from './Home';
export {Login} from './Login';
export {PrivateRoute} from './PrivateRoute';
export {Dashboard} from './Dashboard';
export {Dashboards} from './Dashboards';
export {Report} from './Report';
export {Reports} from './Reports';
