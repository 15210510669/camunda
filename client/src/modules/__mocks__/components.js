export {default as Button} from '../components/__mocks__/Button';
export {default as Input} from '../components/__mocks__/Input';
export {default as Modal} from '../components/__mocks__/Modal';
export {default as Popover} from '../components/__mocks__/Popover';
export {default as ControlGroup} from '../components/__mocks__/ControlGroup';
export {default as ShareEntity} from '../components/__mocks__/ShareEntity';
export {default as DashboardView} from '../components/__mocks__/DashboardView';
export {default as Icon} from '../components/__mocks__/Icon';
export {default as Dropdown} from '../components/__mocks__/Dropdown';
export {default as Message} from '../components/__mocks__/Message';
export {default as ReportView} from '../components/__mocks__/ReportView';
export {default as Select} from '../components/__mocks__/Select';
export {default as BPMNDiagram} from '../components/__mocks__/BPMNDiagram';
export {default as TargetValueBadge} from '../components/__mocks__/TargetValueBadge';
export {default as Table} from '../components/__mocks__/Table';
export {default as Logo} from '../components/__mocks__/Logo';
export {default as ErrorMessage} from '../components/__mocks__/ErrorMessage';
