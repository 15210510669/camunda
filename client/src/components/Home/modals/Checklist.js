/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';
import {LabeledInput} from 'components';
import './Checklist.scss';
import {t} from 'translation';

export default function Checklist({data, formatter, onChange, selectAll, deselectAll}) {
  const formattedData = data.map(formatter);
  const allSelected = formattedData.every(({checked}) => checked);
  const allDeselected = formattedData.every(({checked}) => !checked);
  return (
    <div className="Checklist">
      {formattedData.length > 1 && (
        <LabeledInput
          ref={input => {
            if (input != null) {
              input.indeterminate = !allSelected && !allDeselected;
            }
          }}
          checked={allSelected}
          type="checkbox"
          label={t('common.selectAll')}
          onChange={({target: {checked}}) => (checked ? selectAll() : deselectAll())}
        />
      )}
      <div className="itemsList">
        {formattedData.map(({id, label, checked}) => (
          <LabeledInput
            key={id}
            type="checkbox"
            checked={checked}
            label={label}
            onChange={({target: {checked}}) => onChange(id, checked)}
          />
        ))}
      </div>
    </div>
  );
}
