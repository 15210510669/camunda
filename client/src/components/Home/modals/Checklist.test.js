/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';
import {shallow} from 'enzyme';

import Checklist from './Checklist';
import {LabeledInput} from 'components';

const props = {
  data: [
    {id: 'id1', name: 'name 1', checked: true},
    {id: 'id2', name: 'name 2', checked: false},
    {id: 'id3', name: 'unauthorized', disabled: true},
  ],
  onChange: jest.fn(),
  selectAll: jest.fn(),
  deselectAll: jest.fn(),
};

it('should match snapshot', () => {
  const node = shallow(<Checklist {...props} />);

  expect(node).toMatchSnapshot();
});

it('should invoke onChange when selecting/deselecting an item', () => {
  const node = shallow(<Checklist {...props} />);

  node
    .find(LabeledInput)
    .at(2)
    .simulate('change', {target: {checked: true}});

  expect(props.onChange).toHaveBeenCalledWith('id2', true);
});

it('should invoke enableAll/disableAll when clicking the Select all checkbox', () => {
  const node = shallow(<Checklist {...props} />);

  node
    .find(LabeledInput)
    .at(0)
    .simulate('change', {target: {checked: true}});

  expect(props.selectAll).toHaveBeenCalled();

  node
    .find(LabeledInput)
    .at(0)
    .simulate('change', {target: {checked: false}});

  expect(props.deselectAll).toHaveBeenCalled();
});

it('should hide selectAll if there is only one item', () => {
  const node = shallow(<Checklist {...props} data={[props.data[0]]} />);

  expect(node).toMatchSnapshot();
});
