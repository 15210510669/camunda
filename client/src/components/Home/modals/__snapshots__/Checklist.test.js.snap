// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should hide selectAll if there is only one item 1`] = `
<div
  className="Checklist"
>
  <div
    className="itemsList"
  >
    <ForwardRef(LabeledInput)
      checked={true}
      key="id1"
      label="name 1"
      onChange={[Function]}
      type="checkbox"
    />
  </div>
</div>
`;

exports[`should match snapshot 1`] = `
<div
  className="Checklist"
>
  <ForwardRef(LabeledInput)
    checked={false}
    label="Select All"
    onChange={[Function]}
    type="checkbox"
  />
  <div
    className="itemsList"
  >
    <ForwardRef(LabeledInput)
      checked={true}
      key="id1"
      label="name 1"
      onChange={[Function]}
      type="checkbox"
    />
    <ForwardRef(LabeledInput)
      checked={false}
      key="id2"
      label="name 2"
      onChange={[Function]}
      type="checkbox"
    />
  </div>
</div>
`;
